catchup/peerSelector.go

```diff
@@ -268,19 +268,19 @@ func (hs *historicStats) push(value int, counter uint64, class peerClass) (avera
 	return bounded
 }
 
-// makePeerSelector creates a peerSelector, given a peersRetriever and peerClass array.
-func makePeerSelector(net peersRetriever, initialPeersClasses []peerClass) *peerSelector {
-	selector := &peerSelector{
+// MakePeerSelector creates a PeerSelector, given a peersRetriever and peerClass array.
+func MakePeerSelector(net peersRetriever, initialPeersClasses []PeerClass) *PeerSelector {
```

Comment:
Outside of this function, is peerClass required? I wonder if we could offer a simplified public constructor to avoid how many thing need to change.

In particular initialRank seems like it shouldn't be exposed.

Reply:
I'll create a new constructor and public struct type to handle this.